qxf2_scheduler/routes.py

```diff
@@ -65,73 +68,89 @@ def index():
-        'interviewer_end_time':interviewer_end_time}        
-        if not interviewer_work_time_slots:            
-            interviewer_work_time_slots.append(interviewer_details)
+@app.route("/<interviewer_id>/interviewer/")
+def read_interviewer_details(interviewer_id):
+    "Displays all the interviewer details"
+    parsed_interviewer_details = []
+    interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).filter(
+        Interviewers.interviewer_id == interviewer_id).values(Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewers.interviewer_id, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+    for each_detail in interviewer_details:
+        interviewer_detail = {
+            'interviewers_name': each_detail.interviewer_name,
+            'interviewers_email': each_detail.interviewer_email,
+            'interviewers_designation': each_detail.interviewer_designation,
+            'interviewers_starttime': each_detail.interviewer_start_time,
+            'interviewers_endtime': each_detail.interviewer_end_time}
+        
+        parsed_interviewer_detail = parse_interviewer_detail(interviewer_detail)
+        if not parsed_interviewer_details:
+            parsed_interviewer_details.append(parsed_interviewer_detail)
         else:
-            append_flag = True
-            for each_interviewer in interviewer_work_time_slots:                
-                if interviewer_details['interviewer_email'] in each_interviewer.values():
-                    append_flag = False
-                    if type(each_interviewer['interviewer_start_time'])==list:
-                        each_interviewer['interviewer_start_time'].append(interviewer_details['interviewer_start_time'])  
-                    else:
-                        each_interviewer['interviewer_start_time'] = [each_interviewer['interviewer_start_time'], interviewer_details['interviewer_start_time']]                   
-                    if type(each_interviewer['interviewer_end_time'])==list:
-                        each_interviewer['interviewer_end_time'].append(interviewer_details['interviewer_end_time'])  
-                    else:
-                        each_interviewer['interviewer_end_time'] = [each_interviewer['interviewer_end_time'], interviewer_details['interviewer_end_time']]
-                     
-                    break
-             
-            if append_flag == True:                
-                interviewer_work_time_slots.append(interviewer_details)
-               
-    return render_template("list-interviewer.html", result=interviewer_work_time_slots)   
-    
+            if interviewer_detail['interviewers_name'] in parsed_interviewer_details[0].values():
```

Comment:
I could not follow the logic here. It looks like this code is checking only the first appended dictionary's values for the name. Shouldn't you be looping through every dictionary in parsed_interviewer_details?

Reply:
I am adding answer in the JIRA ticket